### PR TITLE
Refine broker-install intent validation, Fixes AB#3674205

### PR DESCRIPTION
**Pull request by mohitc1**
## Summary
- Refine broker-install intent validation in the embedded WebView.
- Preserve listing attribution parameters when constructing the launch intent.
- Enable the existing validation flight by default and retain its compatibility fallback.
- Add coverage for supported listings, unsupported requests, trailing slashes, and flight-off behavior.

## Validation
- 110 AzureActiveDirectoryWebViewClient tests passed; zero failures, errors, or skips.
- Command: `gradlew.bat :common:testLocalDebugUnitTest --tests "*AzureActiveDirectoryWebViewClientTest*"`
- Device smoke testing remains pending: Authenticator and Company Portal installation, then return to sign-in.
- Confirm effective flight configuration for the target applications before rollout.

## Tracking
Fixes [AB#3674205](https://dev.azure.com/IdentityDivision/Engineering/_workitems/edit/3674205).

### CODE REVIEW

File: common/src/main/java/com/microsoft/identity/common/internal/ui/webview/AzureActiveDirectoryWebViewClient.java [hunk 1]
```diff
@@ -1753,54 +1730,55 @@ private void launchValidatedBrokerInstallIntent(@NonNull final WebView view,
...
+                ? path.substring(0, path.length() - 1)
+                : path;
+        // In a market: URI the operation is carried by the authority rather than the path.
+        return (HTTPS_SCHEME.equalsIgnoreCase(uri.getScheme())
+                    && GOOGLE_PLAY_STORE_HOST.equalsIgnoreCase(uri.getHost())
+                    && GOOGLE_PLAY_STORE_DETAILS_PATH.equals(normalizedPath))
+                || (MARKET_SCHEME.equalsIgnoreCase(uri.getScheme())
+                    && MARKET_DETAILS_AUTHORITY.equalsIgnoreCase(uri.getHost())
+                    && StringUtil.isNullOrEmpty(normalizedPath));
+    }
```
Comment (cacosta33) [inline common/src/main/java/com/microsoft/identity/common/internal/ui/webview/AzureActiveDirectoryWebViewClient.java:1772]: 🟡 Suggestion: Consider rejecting duplicate `id` query parameters before rebuilding this intent. `getQueryParameter("id")` validates one decoded value, but `intent.getData()` preserves the full original query string, so a URI such as `...?id=com.azure.authenticator&id=com.example.attacker` remains ambiguous to downstream parsers (and the reverse ordering is possible too). Requiring exactly one `id` parameter, then validating that value, keeps the value checked here aligned with the URI passed to Play Store. Please add tests for both duplicate orderings if this is in scope.

Comment (mohitc1) [reply common/src/main/java/com/microsoft/identity/common/internal/ui/webview/AzureActiveDirectoryWebViewClient.java:1772]: Addressed in 2d9608f38: require exactly one decoded `id` key. Tests cover both duplicate orderings, encoded-key variants, and attribution preservation. All 113 WebViewClient tests pass.

**Review by cacosta33 — APPROVED** [review]
